tasks/stakingV2.ts

```diff
@@ -0,0 +1,213 @@
+    .addParam("amount", "Amount to of token to be staked without the token decimals.", undefined, types.float)
+    .addOptionalParam("recipient", "Address or contract name that will receive the withdrawn tokens.", undefined, types.string)
+    .addOptionalParam(
+        "fee",
+        "True if withdraw fee to be taken from the amount. False if received amount to equal with withdraw amount.",
+        true,
+        types.boolean,
+    )
+    .addOptionalParam(
+        "cooldown",
+        "False if not exiting from a previous cooldown. True if previous cooldown to be ended.",
+        false,
+        types.boolean,
+    )
+    .addOptionalParam("speed", "Defender Relayer speed param: 'safeLow' | 'average' | 'fast' | 'fastest'", "average", types.string)
+    .setAction(async (taskArgs, hre) => {
+        const signer = await getSigner(hre, taskArgs.speed, false)
+        const chain = getChain(hre)
+
+        const tokenType = taskArgs.asset === "MTA" ? "vault" : "address"
+        const stakingTokenAddress = resolveAddress(taskArgs.asset, chain, tokenType)
+        const stakingToken = StakedToken__factory.connect(stakingTokenAddress, signer)
+        const withdrawAmount = simpleToExactAmount(taskArgs.amount)
+        const recipientAddress = taskArgs.recipient ? resolveAddress(taskArgs.recipient, chain) : await signer.getAddress()
+        const tx = await stakingToken.withdraw(withdrawAmount, recipientAddress, taskArgs.fee, taskArgs.fee)
+        await logTxDetails(tx, `Withdraw ${taskArgs.amount} ${taskArgs.symbol}`)
+    })
+task("staked-withdraw").setAction(async (_, __, runSuper) => {
+    await runSuper()
+})
+
+subtask("staked-claim", "Claim MTA rewards from V2 staking token")
+    .addOptionalParam("recipient", "Address or contract name that will receive the MTA rewards.", undefined, types.string)
+    .addOptionalParam("speed", "Defender Relayer speed param: 'safeLow' | 'average' | 'fast' | 'fastest'", "average", types.string)
+    .setAction(async (taskArgs, hre) => {
+        const signer = await getSigner(hre, taskArgs.speed, false)
+        const chain = getChain(hre)
+
+        const stakingTokenAddress = resolveAddress("MTA", chain, "vault")
+        const stakingToken = StakedToken__factory.connect(stakingTokenAddress, signer)
+        let tx
+        if (taskArgs.recipient) {
+            const recipientAddress = taskArgs.recipient ? resolveAddress(taskArgs.recipient, chain) : await signer.getAddress()
+            tx = await stakingToken["claimReward(address)"](recipientAddress)
+        } else {
+            tx = await stakingToken["claimReward()"]()
+        }
+        const receipt = await logTxDetails(tx, `Claim earned MTA rewards`)
+        console.log(`Claimed ${formatUnits(receipt.events[0].args[2])} MTA rewards`)
+    })
+task("staked-claim").setAction(async (_, __, runSuper) => {
+    await runSuper()
+})
+
+subtask("staked-compound", "Stake any earned MTA rewards")
```

Comment:
No need to have a task for this, since it is not an administrative fn?

Reply:
Oh i see, for fork usage?

Reply:
Yes, used for impersonating users for Support